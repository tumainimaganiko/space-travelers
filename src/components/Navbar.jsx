import { NavLink } from 'react-router-dom';
import Img from '../Images/space-img.png';
import styles from '../styles/Navbar.module.css';

const Navbar = () => {
  const imgStyle = {
    width: '40px',
    height: '40px',
  };

  const line = {
    color: 'black',
    marginLeft: '10px',
    width: '0.2em',
  };

  const activeLink = {
    textDecoration: 'underline',
    color: 'blue',
    fontWeight: '500',
  };

  const link = {
    textDecoration: 'none',
    color: 'blue',
    fontWeight: '500',
  };
  return (
    <div className={styles.container}>
      <div className={styles.heading}>
        <img src={Img} style={imgStyle} alt="space logo" />
        <h1>Space Travelers&apos; Hub</h1>
      </div>
      <ul className={styles.list}>
        <li>
          <NavLink
            to="/"
            style={({ isActive }) => (isActive ? activeLink : link)}
          >
            Rockets
          </NavLink>
        </li>
        <li>
          <NavLink
            to="missions"
            style={({ isActive }) => (isActive ? activeLink : link)}
          >
            Missions
          </NavLink>
          <span style={line}>|</span>
        </li>
        <li>
          <NavLink
            to="myprofile"
            style={({ isActive }) => (isActive ? activeLink : link)}
          >
            My profile
          </NavLink>
        </li>
      </ul>
    </div>
  );
};

export default Navbar;
