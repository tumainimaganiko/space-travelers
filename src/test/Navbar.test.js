import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

it('Render a Navigation Bar', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
