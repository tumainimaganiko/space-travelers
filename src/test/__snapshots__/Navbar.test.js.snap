// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a Navigation Bar 1`] = `
<div
  className="container"
>
  <div
    className="heading"
  >
    <img
      alt="space logo"
      src="space-img.png"
      style={
        Object {
          "height": "40px",
          "width": "40px",
        }
      }
    />
    <h1>
      Space Travelers' Hub
    </h1>
  </div>
  <ul
    className="list"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
            "fontWeight": "500",
            "textDecoration": "underline",
          }
        }
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
            "fontWeight": "500",
            "textDecoration": "none",
          }
        }
      >
        Missions
      </a>
      <span
        style={
          Object {
            "color": "black",
            "marginLeft": "10px",
            "width": "0.2em",
          }
        }
      >
        |
      </span>
    </li>
    <li>
      <a
        className=""
        href="/myprofile"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
            "fontWeight": "500",
            "textDecoration": "none",
          }
        }
      >
        My profile
      </a>
    </li>
  </ul>
</div>
`;
